crawler/crawling/distributed_scheduler.py

```diff
@@ -541,6 +542,11 @@ def next_request(self):
             # extra check to add items to request
             if 'useragent' in req.meta and req.meta['useragent'] is not None:
                 req.headers['User-Agent'] = req.meta['useragent']
+            if 'cookie' in req.meta and req.meta['cookie'] is not None:
```

Comment:
Can you point out in the `master` branch where this was originally possible?

This code block appears to be a duplicate of the lines at https://github.com/istresearch/scrapy-cluster/pull/205/files#diff-0115f1582b20904902916fb395d8fb0fR573.

On the surface, if the request is generated by a spider and has the cookies associated, it will be parsed in the `request_from_dict()` scrapy method. If the request is from a feed item, it will be parsed below. thus, this appears to be duplicate code.

Reply:
Okay so this section is only needed if you are passing the cookie via a `meta` field, instead of passing it in the `cookies` parameter of the request. For example, you can comment out this block and the following unit test will pass

```
        # test request from serialized request with supplied cookie
        exist_req = Request('http://ex.com', cookies={'auth':'101'})
        exist_item = request_to_dict(exist_req)
        self.scheduler.find_item = MagicMock(return_value=exist_item)
        out = self.scheduler.next_request()
        self.assertEqual(out.url, 'http://ex.com')
        for key in out.meta:
            self.assertEqual(out.meta[key], self.req.meta[key])
        self.assertEqual(out.cookies, exist_req.cookies)
        self.req.meta['cookie'] = None  # reset
```

I am going to leave this block in because it enables us to pass the cookie via the `meta` fields as well, that is

Reply:
My commit is at https://github.com/istresearch/scrapy-cluster/commit/15dc714c386607b9dab0932ea1a30530346697ae